docs/src/main/asciidoc/property-source-config.adoc

```diff
@@ -578,17 +595,30 @@ the `Secret` named `s1` would be looked up in the namespace that the application
 See <<namespace-resolution,namespace-resolution>> to get a better understanding of how the namespace
 of the application is resolved.
 
+<<config-map-fail-fast,Similar to the `ConfigMaps`>>; if you want your application to fail to start 
+when it is unable to load `Secrets` property sources, you can set `spring.cloud.kubernetes.secrets.fail-fast=true`.
```

Comment:
I am sorry, but this looks very confusing to me. We say: "if I can't read a Secret and `spring.cloud.kubernetes.secrets.fail-fast=true`, we will fail without retrying". So let's take a step back and see what we do now, without this PR in place: 

- if we can't read a Secret, we will ignore this Exception and continue to start the application. 

This is what would happen if  `spring.cloud.kubernetes.secrets.fail-fast=false` (which is the default), which is good because it means we preserve backwards compatibility in user expectations. 

Now if we enable `spring.cloud.kubernetes.secrets.fail-fast=true` and we can't read a property, we fail - which is again very nice and without surprises.

Now, we want to enable retry and for that we say:

- `spring.cloud.kubernetes.secrets.fail-fast=true`
- put retry and aop on classpath. 

Isn't this confusing to you? The different between retry or not is something on the classpath. _Both_ cases mandate the presence of `spring.cloud.kubernetes.secrets.fail-fast=true`, let me emphasize the `fail-fast` in the naming, which, depending on what you have on the classpath is _not_ a "fail fast"?

As a user of this I would expect a `fail-fast` with `true/false` and a `retry` with `true/false`. 

I am missing something?  

---

This btw has some more implications that I have not yet reviewed entirely. For example the: `ConditionalOnConfigMapPropertiesRetryEnabled`. Look at the name `...RetryEnabled` with a `name = "fail-fast", havingValue = "true"`.  As an exercise (to see if I am not the real problem here), I explained this to 2 of my colleagues at work and they had the same confusion. 

I bet there is something I miss here, so please advise.

Reply:
I have been thinking about this and as a matter of fact found a real case in our repository.

We have `spring-cloud-kubernetes` and `spring-retry` in the same project. If we merge this the way it is and for whatever reason I would like to say "I'd like to fail if I can't read a ConfigMap without ever retrying", I will be stuck as I can't do that. Essentially:

- I already have `spring-retry` and `aop` on the classpath because of business implications.
- I would specify  `spring.cloud.kubernetes.secrets.fail-fast=true`

This case will indeed be documented, no question about it. At the same time, I can't opt out of it. 

Unless there is an explicit flag to control both the fail and retries, someone might not like this.

Reply:
Adding a separate flag for retry was something I've considered during the development. Spring Cloud Config Client and Spring Cloud Kubernetes Config do pretty much the same thing. One loads config from a config server, the other loads them from the Kubernetes objects. And I would expect them to bahave as similar as possible as a user. So I ended up sticking with a very similar design to the config client to keep the ecosystem consistent. The name "fail-fast" also comes from there :) I don't know, I may be wrong with this approach. Btw I don't think that adding one more flag is such a big deal and will create a huge inconsistency. IMHO, we can still change the design unless we get far away from the way other modules does the similar things. What do you think @ryanjbaxter ?